### ISSUE TITLE
Documentation on azurerm_kubernetes_cluster identity block says identity_ids is optional but in fact it's not

### ISSUE THREAD

**Issue by sandy-wang** [hashicorp/terraform-provider-azurerm#17990]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

This impacts when user tries to specify identity type to SystemAssigned when creating aks cluster

### Terraform Version

1.2.6

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks-cluster-dev" {
  name                = "aks-cluster-dev"
  location            = "australiaeast"
  resource_group_name = data.azurerm_resource_group.rg-aks-dev-australiaeast.name

  kubernetes_version                  = "1.22.11"
  role_based_access_control_enabled   = "true"

  default_node_pool {
    name                = "agentpool"
    type                = "VirtualMachineScaleSets"
    enable_auto_scaling = "true"
    min_count           = "1"
    max_count           = "2"
    vm_size             = "Standard_B2s"
    os_disk_size_gb     = "30"
    os_disk_type        = "Managed"
    vnet_subnet_id      = data.azurerm_subnet.aks-dev.id
  }

  identity {
    type = "SystemAssigned"
  }

  kubelet_identity {
    client_id                 = azurerm_user_assigned_identity.mi-aks.client_id
    object_id                 = azurerm_user_assigned_identity.mi-aks.principal_id
    user_assigned_identity_id = azurerm_user_assigned_identity.mi-aks.id
  }

  network_profile {
    network_plugin     = "azure"
    load_balancer_sku  = "standard"
    outbound_type      = "loadBalancer"
  }

}
```

### Debug Output/Panic Output

```shell
STDERR:
    Error: Missing required argument
  
      with azurerm_kubernetes_cluster.aks-cluster-dev,
      on az_aks.tf line 41, in resource "azurerm_kubernetes_cluster" "aks-cluster-dev":
      26:     client_id                 = azurerm_user_assigned_identity.mi-aks.client_id
  
    "kubelet_identity.0.client_id": all of
    `identity.0.identity_ids,kubelet_identity.0.client_id,kubelet_identity.0.object_id,kubelet_identity.0.user_assigned_identity_id`
    must be specified
  
    Error: Missing required argument
  
      with azurerm_kubernetes_cluster.aks-cluster-dev,
      on az_aks.tf line 42, in resource "azurerm_kubernetes_cluster" "aks-cluster-dev":
      27:     object_id                 = azurerm_user_assigned_identity.mi-aks.principal_id
  
    "kubelet_identity.0.object_id": all of
    `identity.0.identity_ids,kubelet_identity.0.client_id,kubelet_identity.0.object_id,kubelet_identity.0.user_assigned_identity_id`
    must be specified
  
    Error: Missing required argument
  
      with azurerm_kubernetes_cluster.aks-cluster-dev,
      on az_aks.tf line 43, in resource "azurerm_kubernetes_cluster" "aks-cluster-dev":
      28:     user_assigned_identity_id = azurerm_user_assigned_identity.mi-aks.id
  
    "kubelet_identity.0.user_assigned_identity_id": all of
    `identity.0.identity_ids,kubelet_identity.0.client_id,kubelet_identity.0.object_id,kubelet_identity.0.user_assigned_identity_id`
    must be specified
```

### Expected Behaviour

When identity type is SystemAssigned, do not require identity_ids. As what's said in documentation.

An identity block supports the following:

[type](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#type) - (Required) Specifies the type of Managed Service Identity that should be configured on this Kubernetes Cluster. Possible values are SystemAssigned, UserAssigned, SystemAssigned, UserAssigned (to enable both).

[identity_ids](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#identity_ids) - (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this Kubernetes Cluster.

NOTE:
This is required when type is set to UserAssigned or SystemAssigned, UserAssigned.

I had to add **identity_ids=[]** to make the terraform apply work

  identity {
    type = "SystemAssigned"
    identity_ids = []
  }

### Actual Behaviour

I had looked at the source code,

https://github.com/hashicorp/terraform-provider-azurerm/blob/773c3e9b79384174519f02a3edd18b6c9174d2cb/internal/services/containers/kubernetes_cluster_resource.go#L240

From L240 - 285, "identity.0.identity_ids" is wrongly referenced in kubelet_identity block.

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#identity
https://github.com/hashicorp/terraform-provider-azurerm/blob/773c3e9b79384174519f02a3edd18b6c9174d2cb/internal/services/containers/kubernetes_cluster_resource.go#L240

**Comment by sandy-wang** [comment]
After moving further with terraform apply and seeing more errors down the track, turned out if bring your own kubelet_identity, the cluster identity type has to be UserAssigned with your own identity provided. Therefore "identity.0.identity_ids" is required. I wish Azure documentation and Terraform documentation could address this clearly, explaining if kubelet identity is user assigned, service identity has to be user assigned too.

**Comment by Amier3** [comment]
Hey @sandy-wang 

Thanks for bringing this to our attention! We'll get the fix (#18027) merged in soon and fix this :)
